Headlines/AppDelegate.swift

```diff
@@ -127,35 +127,33 @@ class AppDelegate: UIResponder,
         hud.textLabel.text = "Loading"
         hud.show(in: vc.view)
 
-        let success: (([News]?) -> Void) = { (news) in
-            hud.dismiss()
+        let handler: ((Result <[News], Error>) -> Void) = { result in
+            switch result {
+            case .success(let news):
+                hud.dismiss()
 
-            guard let news = news else {
-                return
-            }
-
-            let storyboard = UIStoryboard.init(name: "News", bundle: Bundle.main)
-            guard let newsViewController = storyboard.instantiateViewController(withIdentifier: "news")
-                as? NewsTableViewController else {
-                return
-            }
-            newsViewController.preferredDateStyle = .short
-            newsViewController.trackContextFrom = .search
-            newsViewController.news = news
-            newsViewController.title = topic
+                let storyboard = UIStoryboard.init(name: "News", bundle: Bundle.main)
+                guard let newsViewController = storyboard.instantiateViewController(withIdentifier: "news")
+                    as? NewsTableViewController else {
+                        return
+                }
+                newsViewController.preferredDateStyle = .short
+                newsViewController.trackContextFrom = .search
+                newsViewController.news = news
+                newsViewController.title = topic
 
-            let navController = DismissableNavigationController(rootViewController: newsViewController)
-            newsViewController.navigationItem.leftBarButtonItem = navController.dismissButtonItem
+                let navController = DismissableNavigationController(rootViewController: newsViewController)
+                newsViewController.navigationItem.leftBarButtonItem = navController.dismissButtonItem
 
-            vc.present(navController, animated: true, completion: nil)
-        }
+                vc.present(navController, animated: true, completion: nil)
 
-        let fail: ((NSError) -> Void) = { (error) in
-            hud.dismiss()
+            case .failure(_):
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/cafeb7b8636cbafd0216eaa2489bb108aa52959c/Headlines/AppDelegate.swift#L150"></span>Arguments can be omitted when matching enums with associated types if they are not used.<br />
<code>empty_enum_arguments</code> <code>AppDelegate.swift:150</code></td>
    </tr>
  </tbody>
</table>